### PR TITLE
N21 1264 new class page extension

### PR DESCRIPTION
**Pull request by IgorCapCoder**
# Short Description
To replace the functionality of the old classes table, the action icons for classes (not groups) are added

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/N21-1264
https://github.com/hpi-schul-cloud/schulcloud-client/pull/3335
https://github.com/hpi-schul-cloud/schulcloud-server/pull/4474

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/locales/de.json [hunk 1]
```diff
@@ -737,6 +737,8 @@
 	"pages.administration.school.index.authSystems.delete": "{system} löschen",
 	"pages.administration.classes.index.title": "Klassen verwalten",
 	"pages.administration.classes.index.add": "Klasse hinzufügen",
+	"pages.administration.classes.deleteDialog.title": "Klasse löschen?",
+	"pages.administration.classes.deleteDialog.content": "\"Möchten Sie wirklich die Klasse '<i>{itemName}</i>' löschen?\"",
```
Comment (arnegns) [inline src/locales/de.json:741]: The sentence structure doesn't look right

File: src/pages/administration/ClassOverview.page.unit.ts [hunk 2]
```diff
@@ -8,13 +8,26 @@ import createComponentMocks from "@@/tests/test-utils/componentMocks";
 import Vue from "vue";
 import { SortOrder } from "@/store/types/sort-order.enum";
 import { Pagination } from "@/store/types/commons";
+import setupStores from "../../../tests/test-utils/setupStores";
+import AuthModule from "../../store/auth";
+import { authModule } from "../../store";
```
Comment (arnegns) [inline src/pages/administration/ClassOverview.page.unit.ts:13]: Check your imports

File: src/pages/administration/ClassOverview.page.unit.ts [hunk 3]
```diff
@@ -202,4 +223,257 @@ describe("ClassOverview", () => {
...
+			it("should open the delete dialog", async () => {
+				const { wrapper } = setup();
+
+				await wrapper
+					.find('[data-testid="class-table-delete-icon"]')
+					.trigger("click");
+
+				const dialog = wrapper.find('[data-testid="delete-dialog"]');
+
+				expect(dialog.exists()).toBeTruthy();
```
Comment (arnegns) [inline src/pages/administration/ClassOverview.page.unit.ts:430]: You should check for the prop on your component instead because the dialog also exists in the dom but is hidden

File: src/pages/administration/ClassOverview.page.vue [hunk 4]
```diff
@@ -22,8 +22,88 @@
 			@update:items-per-page="onUpdateItemsPerPage"
 			@update:page="onUpdateCurrentPage"
 		>
+			<template v-slot:[`item.actions`]="{ item }">
+				<v-icon
+					v-if="hasPermission && !item.externalSourceName"
```
Comment (arnegns) [inline src/pages/administration/ClassOverview.page.vue:27]: you could move this v-if logic to script block

File: src/pages/administration/ClassOverview.page.vue [hunk 5]
```diff
@@ -22,8 +22,88 @@
...
+						@click="onCloseDeleteDialog"
+					>
+						{{ t("common.actions.cancel") }}
+					</v-btn>
+					<v-btn
+						data-testId="dialog-confirm"
+						class="px-6"
+						color="primary"
+						depressed
+						@click="onDeleteClass"
```
Comment (arnegns) [inline src/pages/administration/ClassOverview.page.vue:99]: You should use the same name convention for both buttons.. so i would suggest onConfirmClassDeletion

File: src/pages/administration/ClassOverview.page.vue [hunk 6]
```diff
@@ -22,8 +22,88 @@
...
+					data-testid="class-table-manage-icon"
+					@click="manageClass(item)"
+				>
+					{{ mdiAccountGroupOutline }}
+				</v-icon>
+				<v-icon
+					v-if="hasPermission && !item.externalSourceName"
+					size="small"
+					data-testid="class-table-edit-icon"
+					@click="editItem(item)"
```
Comment (arnegns) [inline src/pages/administration/ClassOverview.page.vue:38]: I dont like the name editItem. The variable name is item but what you are editing?

File: src/pages/administration/ClassOverview.page.vue [hunk 7]
```diff
@@ -22,8 +22,88 @@
...
+				>
+					{{ mdiPencilOutline }}
+				</v-icon>
+				<v-icon
+					v-if="hasPermission && !item.externalSourceName"
+					size="small"
+					data-testid="class-table-delete-icon"
+					@click="onOpenDeleteDialog(item)"
```
Comment (arnegns) [inline src/pages/administration/ClassOverview.page.vue:46]: the name should be deleteXYZ. The on convention regards to the events

File: src/store/group.unit.ts [hunk 8]
```diff
@@ -257,4 +259,63 @@ describe("GroupModule", () => {
 			});
 		});
 	});
+
+	// TODO test deleteClass method by mocking the v1 axios call
```
Comment (arnegns) [inline src/store/group.unit.ts:263]: Todo?

File: src/store/group.unit.ts [hunk 9]
```diff
@@ -17,6 +17,8 @@ import GroupModule from "./group";
 import { createMock, DeepMocked } from "@golevelup/ts-jest";
 import { mockApiResponse } from "@@/tests/test-utils/mockApiResponse";
 import { mapAxiosErrorToResponseError } from "@/utils/api";
+import { classInfoFactory } from "../../tests/test-utils/factory/classInfoFactory";
```
Comment (arnegns) [inline src/store/group.unit.ts:20]: use @@/tests/..

File: src/store/group.unit.ts [hunk 10]
```diff
@@ -257,4 +259,63 @@ describe("GroupModule", () => {
...
+				expect(axios.delete).toHaveBeenCalled();
+			});
+
+			it("should load classes", async () => {
+				const { classId } = setup();
+
+				await module.deleteClass(classId);
+
+				expect(module.loadClassesForSchool).toHaveBeenCalled();
+				expect(module.getClasses).toEqual([]);
```
Comment (arnegns) [inline src/store/group.unit.ts:289]: the first expect should be enough

**Review by arnegns — CHANGES_REQUESTED** [review]

File: src/pages/administration/ClassOverview.page.unit.ts [hunk 11]
```diff
@@ -202,4 +223,257 @@ describe("ClassOverview", () => {
 			});
 		});
 	});
+
```
Comment (MarvinOehlerkingCap) [inline src/pages/administration/ClassOverview.page.unit.ts:226]: this component test is missing tests for displaying items in the table and displaying an empty state

Comment (MarvinOehlerkingCap) [reply src/pages/administration/ClassOverview.page.unit.ts:226]: PS: if you don't want to test the table component itself, then at least test if you used the api of the component correctly by checking if the props are set correctly

**Review by MarvinOehlerkingCap — CHANGES_REQUESTED** [review]

File: src/locales/de.json (hunk 1 above)
Comment (IgorCapCoder) [reply src/locales/de.json:741]: after analyzing all locales doublequotes are used most often. But single quotes and other options are also used more than once. Maybe this could be cleaned up in the future :)

**Review by MarvinOehlerkingCap — DISMISSED** [review]

**Review by arnegns — DISMISSED** [review]

**Review by MarvinOehlerkingCap — APPROVED** [review]

**Review by MarvinOehlerkingCap — APPROVED** [review]
